caikit/runtime/utils/import_util.py

```diff
@@ -119,6 +119,14 @@ def get_data_model(config: aconfig.Config = None) -> UnifiedDataModel:
     for lib_name in base_lib_names:
         cdm = _get_cdm_from_lib(lib_name, cdm)
 
+    # Ensure that all model management components have been initialized
+    # TODO: This function has a _ton_ of side effects! We need to split it up
+    #   and isolate these side effects in appropriately named functions.
+    #   Specifically, this function is not only responsible for creating the
+    #   data model, but it performs the dynamic import of the domain library and
+    #   initializes the model management components.
+    MODEL_MANAGER.initialize_components()
```

Comment:
Can we start that breakup by not putting this here?

Totally agree with this comment as well- the service factory calls this method just to get the dynamic library import, and then the further calls to this method from the actual servicers are, as far as I can tell, unused or unneeded. I don't think we ever need to actually use the `cdm` for anything.

Reply:
I'm going to punt this to a future issue for now to avoid blowing up this PR